lib/core/src/display/shape/text/glyph/font.rs

```diff
@@ -203,27 +199,28 @@ impl FontRegistry {
         FontRegistry {
             embedded : EmbeddedFonts::create_and_fill(),
             fonts    : HashMap::new(),
-            next_id  : 1
         }
     }
 
-    /// Load data from one of embedded fonts. Returns None if embedded font not found.
-    pub fn load_embedded_font(&mut self, name:&str) -> Option<FontId> {
-        let render_info = FontRenderInfo::from_embedded(&self.embedded,name);
-        render_info.map(|info| self.put_render_info(info))
-    }
-
-    fn put_render_info(&mut self, font:FontRenderInfo) -> FontId {
-        let id = self.next_id;
-        self.fonts.insert(id,font);
-        self.next_id += 1;
-        id
+    /// Get render font info from loaded fonts, and if it does not exists, load data from one of
+    /// embedded fonts. Returns None if the name is missing in both loaded and embedded font list.
+    pub fn get_or_load_embedded_font(&mut self, name:&str) -> Option<FontHandle> {
+        match self.fonts.entry(name.to_string()) {
+            Occupied(entry) => Some(entry.get().clone()),
+            Vacant(entry)   => match FontRenderInfo::from_embedded(&self.embedded,name) {
+                Some(render_info) => {
+                    let rc = Rc::new(render_info);
+                    entry.insert(rc.clone());
+                    Some(rc)
+                },
+                None => None
+            }
+        }
     }
 
-    /// Get render info of one of loaded fonts. Panics for unrecognized id - you should only use
-    /// ids returned from `Fonts`' functions.
-    pub fn get_render_info(&mut self, id:FontId) -> &mut FontRenderInfo {
-        self.fonts.get_mut(&id).unwrap()
+    /// Get render info of one of loaded fonts.
+    pub fn get_render_info(&mut self, name:&str) -> Option<FontHandle> {
```

Comment:
The docs and name tells about render info. But it returns `FontHandle` now. I dont see any connection between those two now.